helix-term/src/application.rs

```diff
@@ -331,6 +332,11 @@ impl Application {
 
         self.event_loop().await;
 
+        future::join_all(self.editor.language_servers.iter_clients().map(|client| {
+            tokio::time::timeout(Duration::from_millis(500), client.force_shutdown())
```

Comment:
I think you can just time out the join_all future here

Reply:
Done.